parser/src/powdr.lalrpop

```diff
@@ -622,19 +621,16 @@ MatchArm: MatchArm = {
 
 IfExpression: Box<Expression> = {
     <start:@L> "if" <condition:BoxedExpression>
-        <body:BracedExpression>
+        <body:BlockExpression>
         "else"
-        <else_body:BracedExpression>
+        <else_body:BlockExpression>
         <end:@R> => ctx.to_expr_with_source_ref(IfExpression{condition, body, else_body}, start, end),
 }
 
-BlockExpression: Box<Expression> = {
-    <start:@L> "{" <statements:StatementInsideBlock+> <expr:BoxedExpression> "}" <end:@R> => ctx.to_expr_with_source_ref(BlockExpression{statements, expr}, start, end),
-}
 
-BracedExpression: Box<Expression> = {
-    <start:@L> "{" <statements:StatementInsideBlock+> <expr:BoxedExpression> "}" <end:@R> => ctx.to_expr_with_source_ref(BlockExpression{statements, expr}, start, end),
-    <start:@L> "{" <expr:BoxedExpression> "}" <end:@R> => ctx.to_expr_with_source_ref(BlockExpression{statements: vec![], expr}, start, end),
+BlockExpression: Box<Expression> = {
+    <start:@L> "{" <statements:StatementInsideBlock*> "}" <end:@R> => ctx.to_expr_with_source_ref(BlockExpression{statements, expr: Box::new(Expression::Tuple(ctx.source_ref(start, end), vec![]))}, start, end),
```

Comment:
I'm not sure I would implement the tuple at the syntax level. Is it much more difficult to make `expr` and `Option<Expression>`?

Reply:
Also the source reference is misleading.